### DISCUSSION TITLE
Possible to take a screenshot?

### DISCUSSION THREAD

**Discussion by oulfik** [jeffreylanters/react-unity-webgl#232] [Question and Answer] [answered]
Is it possible to take a screenshot with react-unity-webgl or do I have to call the unity function _ScreenCapture.CaptureScreenshot()_ for this purpose?

**Comment by jeffreylanters** [comment, marked as answer]
Hi, ScreenCapture.CaptureScreenshot is not available is WebGL. I will add support for taking screenshots in the near future. But for now, there is an easy solution for taking screenshots from your web application as following:

```js
import React, { useEffect, useState } from "react";
import Unity, { UnityContext } from "react-unity-webgl";

const unityContext = new UnityContext({
  loaderUrl: "build/myunityapp.loader.js",
  dataUrl: "build/myunityapp.data",
  frameworkUrl: "build/myunityapp.framework.js",
  codeUrl: "build/myunityapp.wasm",
});

function App() {
  const [canvas, setCanvas] = useState(null);

  useEffect(function () {
    unityContext.on("canvas", setCanvas);
  }, []);

  function takeScreenshot () {
    const screenshotBase64 = canvas.toDataURL();
    // Do something with the screenshot data...
  }

  return (
    <div>
      <button onClick={takeScreenshot}>Take Screenshot</button>
      <Unity unityContext={unityContext} />
    </div>
  );
}
```

**Reply by oulfik** [reply]
Thanks! Works like a charm. One thing to consider: I had to change one line of code in the loader.js file:
`preserveDrawingBuffer: true`
Before that I only got a blank image. (See [link](https://www.blend4web.com/en/forums/topic/3302/))

**Reply by jeffreylanters** [reply]
Awesome! Thanks for the feedback, I'll keep that in mind while implementing this natively.

**Reply by jeffreylanters** [reply]
FYI, the option to enable `preserveDrawingBuffer` is added in [release v8.5.1](https://github.com/jeffreylanters/react-unity-webgl/releases/tag/v8.5.1).

**Reply by jeffreylanters** [reply]
FYI, support for taking screenshots has ben added natively in [release v8.6.0](https://github.com/jeffreylanters/react-unity-webgl/releases/tag/v8.6.0).
